### PR TITLE
add basic jxlinfo binary

### PR DESCRIPTION
**Pull request by mo271**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 debug = true
 
 [workspace]
-members = [ "jxl", "jxl_cli", "jxl_headers_derive"]
+members = [ "jxl", "jxl_cli", "jxl_headers_derive", "jxlinfo"]
```
Comment (veluca93) [inline Cargo.toml:5]: Why is it not part of `jxl_cli`? You can have more than one binary per crate

Comment (mo271) [reply Cargo.toml:5]: ok, will put it there

Comment (mo271) [reply Cargo.toml]: moved, now it runs with 
```
cargo run -p jxl_cli --bin jxlinfo [your filename here]
```

File: jxl_cli/src/bin/jxlinfo.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+use std::fs;
+use std::io::Read;
+
+fn parse_jxl_codestream(data: &[u8], verbose: bool) -> Result<(), jxl::error::Error> {
+    let mut br = BitReader::new(data);
+    let fh = FileHeaders::read(&mut br)?;
+
+    // Non-verbose output
+    if !verbose {
+        // let color_space = &fh.image_metadata.color_encoding;
```
Comment (veluca93) [inline jxl_cli/src/bin/jxlinfo.rs:21]: leftover?

File: jxl_cli/src/bin/jxlinfo.rs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+        // TODO(firsching): add non-verbose print for bit-depth
+        // TODO(firsching): add non-verbose print for color space
+        return Ok(());
+    }
+
+    // Verbose output: Use Debug trait to print the FileHeaders
+    println!("{:#?}", fh);
+
+    let _icc = if fh.image_metadata.color_encoding.want_icc {
```
Comment (veluca93) [inline jxl_cli/src/bin/jxlinfo.rs:38]: Why are you making a binding for the result here?

File: jxl_cli/src/bin/jxlinfo.rs (hunk 2 above)
Comment (mo271) [reply jxl_cli/src/bin/jxlinfo.rs]: yes, removed

File: jxl_cli/Cargo.toml [hunk 4]
```diff
@@ -5,6 +5,7 @@ edition = "2021"
 
 [dependencies]
 jxl = { path = "../jxl" }
+clap = { version = "4.5.18", features = ["derive"] }
```
Comment (veluca93) [inline jxl_cli/Cargo.toml:8]: I don't think you're actually using the derive feature...

**Review by veluca93 — APPROVED** [review]

File: jxl_cli/Cargo.toml (hunk 4 above)
Comment (mo271) [reply jxl_cli/Cargo.toml]: you are right: removed it

**Pull request merged**